cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -966,6 +1055,56 @@ public synchronized void setFlowletInstances(HttpRequest request, HttpResponder
     }
   }
 
+  /**
+   * Update the log level for a flowlet according to the request body. Currently supported program types are Flow,
+   * Worker and Service. The request body is expected to contain a map of log levels, where key is loggername, value is
+   * one of the valid {@link org.apache.twill.api.logging.LogEntry.Level} or null.
+   */
+  @PUT
+  @Path("/apps/{app-id}/flows/{flow-id}/flowlets/{flowlet-id}/logLevels")
+  public synchronized void updateFlowletLogLevels(HttpRequest request, HttpResponder responder,
+                                                  @PathParam("namespace-id") String namespaceId,
+                                                  @PathParam("app-id") String appId,
+                                                  @PathParam("flow-id") String flowId,
+                                                  @PathParam("flowlet-id") String flowletId) throws Exception {
+    try {
+      // we are decoding the body to Map<String, String> instead of Map<String, LogEntry.Level> here since Gson will
+      // serialize invalid enum values to null, which is allowed for log level, instead of throw an Exception.
+      lifecycleService.updateProgramLogLevels(new ProgramId(namespaceId, appId, ProgramType.FLOW, flowId),
```

Comment:
to remove code duplication, you should call the same updateProgramLogLevels method that the others call